filemanager.go

```diff
@@ -0,0 +1,358 @@
+			uploads.add(localRelativeName, localFile)
+			continue
+		}
+
+		if localDigest, err := getFileDigest(localFile.Path); err != nil {
+			return nil, err
+		} else if localDigest != remoteFile.Digest {
+			uploads.add(localRelativeName, localFile)
+		}
+	}
+	return &uploads, nil
+}
+
+// snapshotLocal returns a channel which receives the infos of the files under the given basePath.
+func (m *Manager) snapshotLocal(path string) error {
+
+	m.localFiles.files = make(map[string]*fileInfo)
+
+	basePath := filepath.ToSlash(filepath.Clean(path))
+	stat, err := os.Stat(basePath)
+
+	if err != nil {
+		return err
+	}
+
+	if !stat.IsDir() {
+		return fmt.Errorf("the path %s is not a directory", basePath)
+	}
+
+	err = filepath.Walk(basePath, func(p string, stat os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		fileName, err := filepath.Rel(basePath, p)
+		if err != nil {
+			return err
+		}
+
+		fileInfo := fileInfo{
+			File: File{
+				Name:  fileName,
+				Path:  p,
+				Size:  int(stat.Size()),
+				IsDir: stat.IsDir(),
+			},
+		}
+		m.localFiles.add(fileName, &fileInfo)
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listFileManagerFiles returns a channel which receives the infos of the files under the given basePath.
+func (m *Manager) snapshotRemote(ctx context.Context, remotePath string) error {
+
+	m.remoteFiles.files = make(map[string]*fileInfo)
+
+	basePath := path.Clean(remotePath)
+
+	query := ListQuery{
+		ItemsPerPage: 1000,
+		Offset:       0,
+		Search: ListSearch{
+			Path: fmt.Sprintf("^%s/.*", basePath),
+		},
+	}
+
+	for {
+		files, err := m.client.ListFiles(ctx, query)
+		if err != nil {
+			return err
+		}
+		for _, file := range files.Items {
+
+			if err != nil {
+				return err
+			}
+
+			if strings.HasPrefix(file.Path, basePath) && file.Path != basePath {
+				remoteRelativeName := strings.TrimPrefix(file.Path, basePath+"/")
+				m.remoteFiles.add(remoteRelativeName, &fileInfo{File: file})
+			}
+		}
+
+		query.Offset += query.ItemsPerPage
+		if query.Offset > files.Total {
+			break
+		}
+	}
+
+	return nil
+}
+
+// upload uploads the file to the filemanager.
+func (m *Manager) upload(ctx context.Context, file *fileInfo, sourcePath, destPath string) error {
+
+	// We do not upload directories
+	if file.IsDir {
+		return nil
+	}
+
+	baseName := filepath.Base(file.Name)
+	dirName := filepath.ToSlash(filepath.Dir(file.Name))
+	destinationPath := filepath.ToSlash(filepath.Join(destPath, dirName))
+
+	fmt.Printf("Uploading %s to %s\n", file.Path, destinationPath)
+
+	if m.dryrun {
+		return nil
+	}
+
+	reader, err := os.Open(file.Path)
+	if err != nil {
+		return err
+	}
+
+	defer reader.Close()
+
+	if err := m.client.UploadFileReplace(ctx, destinationPath, baseName, true, reader); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// deleteRemote deletes the remote file.
+func (m *Manager) deleteRemote(remotePath string) error {
+
+	println("Deleting", remotePath)
```

Comment:
Please use `fmt.Println`, as `println` is [not considered stable](https://go.dev/ref/spec#Bootstrapping)